### PR TITLE
feat: Social Media Posts CRUD operation and updated basic profile page.

### PR DESCRIPTION
**Pull request by kedark152**
### feat: Social Media Posts CRUD operation and updated basic profile page.

Preview Pages for checking:- Home Page & Profile Page.
Link: https://deploy-preview-3--agrolife.netlify.app/

Features added in PR:-
1. Create New Post, Update Post & Delete Post.
2. Image addition and change post image using edit.
3. Image upload feature using cloudinary API and storage.
4. Sort Posts By Date by clicking Newest and oldest buttons.
5. Toast for success of each CRUD operation.
6. Added emoji picker for react.
7. postModal Slice for CRUD actions and updating state.
8. postSlice for fetching all posts lists and updating state (will implement the feed list in next PR)

### CODE REVIEW

**Review by KanekarSnehal — APPROVED** [review]

**Review by bajajcodes — APPROVED** [review]
- Features are working are as expected.
- Code is good to merge.

File: src/components/EditPostModal.jsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+      postTextContent.concat(emojiObject.emoji)
+    );
+  }
+
+  async function handleOnSubmit(e) {
+    e.preventDefault();
+
+    let imageLink = imageSrc;
+    const formData = new FormData(e.currentTarget);
+    console.log(formData.get('uploadImage'));
```
Comment (rahulcs754) [inline src/components/EditPostModal.jsx:56]: i thin you should remove this console

File: src/components/EditPostModal.jsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  Textarea,
+  ModalFooter,
+  useDisclosure,
+} from '@chakra-ui/react';
```
Comment (rahulcs754) [inline src/components/EditPostModal.jsx:18]: nice to use component

**Review by rahulcs754 — APPROVED** [review]
Nice to write every code.

**Comment by kedark152** [comment]
Thanks all for PR Review, merging PR

**Pull request merged**